core/src/main/scala/cats/Apply.scala

```diff
@@ -216,6 +216,13 @@ trait Apply[F[_]] extends Functor[F] with InvariantSemigroupal[F] with ApplyArit
 }
 
 object Apply {
+
+  /**
+   * This semigroup uses a product operation to combine `F`s,
+   * if the `Apply[F].product` result in larger `F` e.g. when `F` is a `List`,
+   * accumulative usage of this instance, such as `combineAll`, will result in
+   * `F`s with exponentially increasing sizes.
```

Comment:
Light copy editing (spans lines, so I couldn't GitHub it):

```
   * This semigroup uses a product operation to combine `F`s.
   * If the `Apply[F].product` results in larger `F` (i.e. when `F` is a `List`),
   * accumulative usage of this instance, such as `combineAll`, will result in
   * `F`s with exponentially increasing sizes.
```

Reply:
thanks!